src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp

```diff
@@ -676,9 +682,78 @@ void HydrofractureSolver< POROMECHANICS_SOLVER >::assembleSystem( real64 const t
+
   this->getRefDerivativeFluxResidual_dAperture()->zero();
 }
 
+template< typename POROMECHANICS_SOLVER >
+void HydrofractureSolver< POROMECHANICS_SOLVER >::
+assembleFluidLeakSource( double time,
+                         double dt,
+                         DomainPartition & domain,
+                         DofManager const & dofManager,
+                         CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                         arrayView1d< real64 > const & localRhs ) const
+{
+
+  GEOS_MARK_FUNCTION;
+
+  string const presDofKey = dofManager.getKey( SinglePhaseBase::viewKeyStruct::elemDofFieldString() );
+
+  globalIndex const rankOffset = dofManager.rankOffset();
+
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const &,
+                                                                MeshLevel const & mesh,
+                                                                arrayView1d< string const > const & regionNames )
+  {
+    ElementRegionManager const & elemManager = mesh.getElemManager();
+
+    elemManager.forElementSubRegions< FaceElementSubRegion >( regionNames,
+                                                              [&]( localIndex const,
+                                                                   FaceElementSubRegion const & subRegion )
+    {
+      localIndex regionSize = subRegion.size();
+
+      string const & fluidName = subRegion.getReference< string >( FlowSolverBase::viewKeyStruct::fluidNamesString() );
+      SingleFluidBase const & fluid = this->template getConstitutiveModel< SingleFluidBase >( subRegion, fluidName );
+
+      arrayView1d< globalIndex const > const presDofNumber = subRegion.getReference< array1d< globalIndex > >( presDofKey );
+
+      arrayView2d< real64 const > const dens = fluid.density();
+      arrayView2d< real64 const > const dDens_dPressure = fluid.dDensity_dPressure();
+
+      arrayView1d< real64 const > const area = subRegion.getElementArea();
+      arrayView1d< integer const > const elemGhostRank = subRegion.ghostRank();
+
+      arrayView1d< real64 const > const fractureCreationTime = subRegion.getField< fields::flow::fractureCreationTime >();
+
+      constexpr integer numDof = 1;
+      constexpr integer numEqn = 1;
+      globalIndex dofIndices[numDof]{};
+      real64 localJacobian[numEqn][numDof]{};
+      forAll< serialPolicy >( regionSize,
+                              [&] GEOS_HOST_DEVICE ( localIndex const kfe )
+      {
+        if( elemGhostRank[kfe] >= 0 || fabs( time - fractureCreationTime[kfe] ) < 1e-12 )
+          return;
+        const globalIndex localRow = presDofNumber[kfe] - rankOffset;
+        for( integer idof = 0; idof < numDof; ++idof )
+          dofIndices[idof] = presDofNumber[kfe] + idof;
+        localJacobian[0][0] = m_leakoffCoefficient * area[kfe] * dDens_dPressure[kfe][0] / LvArray::math::sqrt( time + dt / 2.0 - fractureCreationTime[kfe] );
```

Comment:
```suggestion
        localJacobian[0][0] = 2.0 * m_leakoffCoefficient * area[kfe] * dDens_dPressure[kfe][0] / LvArray::math::sqrt( time + dt / 2.0 - fractureCreationTime[kfe] );
```
There should be scaler term of 2 for Carter's model?

Reply:
> There should be a scaler term of 2 for Carter's model?

Yeah, theoretically. It also depends how this variable is defined. I originally assume 2 is lumped into the leakoffCoefficient. But i will add explanation to this variable in the code to clarify.